### DISCUSSION TITLE
Significant difference between scoreboard result and test resuls.

### DISCUSSION THREAD

**Discussion by Moohaa** [geeksblabla/blanat#150] [Q&A] [answered]
Hi, My solution typically takes an average of 2.4 minutes to finish processing 1 billion rows. However, on the scoreboard, it shows a duration of 17 minutes. What could be the reason for this significant difference?

**Comment by aboullaite** [comment]
Not sure why it is much faster in your machine than ours, but the runtime is different and there are many things that inherently will be different. All submissions runs within the same machine

**Reply by Moohaa** [reply]
Yes ,weird.I've tried many time and it took  in average (2.4min). 

btw my machine config is:
- cpu Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz.
- ram 16gb.
- java runtime (x86) graalvm-jdk-17.0.8

And my solution only uses java standards library.
So normally it shouldn't be that big of difference.

**Reply by aboullaite** [reply]
Perf wise, our machine is way better. Maybe try openjdk intstead, whoch is the one we are using

**Reply by Moohaa** [reply]
Yes, I Ithought the same.

I've tried  also  to run it in a openjdk environment (jdk-17.0.10_windows-x64_bin). 

It finishes on average in 4.6 minutes (5.5 minutes in the worst case).

**Reply by sqrt-negativeone** [reply]
Could it be that the test machine have an HDD instead of SSD? I'm noticing similar trends; on my machine (i5 12500H, 32gb ddr4 with a ssd drive) my solution does about 3 seconds

![image](https://github.com/geeksblabla/blanat/assets/37900094/adc85f5c-c9ec-4ad7-ab51-9d36727979e9)

**Reply by samir-** [reply]
@sqrt-minus-one i noticed your solution uses 16 threads. the judge machine is an 8vcores one so that must be a first diff. 

3 sec solution is not possible unless your file is already cached into memory. 22go is impossible to load from SSD given the raw hardware speed :) so the only explanation is first run takes much much longer then the next runs benefits from the cached file in memory and that can explain the average but the median is still odd .

**Reply by sqrt-negativeone** [reply]
@samir- Oh you are right I haven't considered the disk caches! after clearing the caches it jumped to 19s which makes sense now :laughing:

**Reply by aboullaite** [reply]
@Moohaa your code now runs under 6min :) it reflects what you found

**Reply by Moohaa** [reply, marked as answer]
Yes, thank you @aboullaite . 
The issue was that I used Runtime.getRuntime().availableProcessors(), and it returned 1 in the judge VM. This is a known problem [https://repost.aws/questions/QUDkXshWl4RiKuTd_qdDMVxw/jvm-runtime-getruntime-availableprocessors-returns-1-on-aws-ecs-fargate](url).
